liquibase-core/src/main/java/liquibase/snapshot/jvm/ForeignKeySnapshotGenerator.java

```diff
@@ -184,14 +184,7 @@ protected DatabaseObject snapshotObject(DatabaseObject example, DatabaseSnapshot
                         row.getInt(METADATA_DELETE_RULE), database);
                 foreignKey.setDeleteRule(deleteRule);
 
-                short deferrability;
-                if (((database instanceof MySQLDatabase) && ((MySQLDatabase) database)
-                    .hasBugJdbcConstraintsDeferrable()) || ((database instanceof MariaDBDatabase) && (
-                        (MariaDBDatabase) database).hasBugJdbcConstraintsDeferrable())
-                   )
-                    deferrability = DatabaseMetaData.importedKeyNotDeferrable;
-                else
-                    deferrability = row.getShort(METADATA_DEFERRABILITY);
+                short deferrability = row.getShort(METADATA_DEFERRABILITY);
```

Comment:
We should also pull out the now-unused hasBugJdbcConstraintsDeferrable() function too